proto/storage/collector_runtime_config.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package storage;
+
+import "google/protobuf/timestamp.proto";
+import "storage/resource_collection.proto";
+import "storage/user.proto";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+message CollectorRuntimeConfigACK {
+  string error = 1;
+  Action action = 2;
+
+  enum Action {
+    ACK = 0;
+    NACK = 1;
+  }
+}
+
+enum RuntimeFeature {
+  PROCESSES = 0;
+  NETWORK_CONNECTIONS = 1;
+  NETWORK_ENDPOINTS = 2;
+}
+
+message Status {
+  message ProcessStatus {
+    bool enabled = 1;
+  }
+
+  message NetworkConnectionStatus {
+    bool enabled = 1;
+    bool aggregate_external = 2;
+  }
+
+  message NetworkEndpointStatus {
+    bool enabled = 1;
+    bool processes_listening_on_port = 2;
+  }
+
+  oneof status {
+    ProcessStatus process_status = 1;
+    NetworkConnectionStatus network_connection_status = 2;
+    NetworkEndpointStatus network_endpoint_status = 3;
+  }
+}
+
+message RuntimeFeatureConfig {
+  message RuntimeRule {
+    string resource_collection_name = 1;
+    Status status = 2;
+    google.protobuf.Timestamp created_at = 3;
+    google.protobuf.Timestamp last_updated = 4;
+    SlimUser created_by = 5;
+    SlimUser updated_by = 6;
+  }
+
+  RuntimeFeature feature = 1;
+  Status default_status = 2;
+  repeated RuntimeRule rules = 3;
+  google.protobuf.Timestamp created_at = 4;
+  google.protobuf.Timestamp last_updated = 5;
+  // The default is not created by a user, but by central so there is no created_by.
+  SlimUser updated_by = 6;
+}
+
+message CollectorRuntimeConfig {
+  repeated RuntimeFeatureConfig runtime_feature_config = 1;
+  repeated ResourceCollection resource_collections = 2;
+}
+
+// Collector needs to know the cluster it is running on to apply the rules correctly.
+message CollectorRuntimeConfigWithCluster {
+  string cluster_name = 1;
```

Comment:
I assume that eventually this data will be stored in a table some place.  So I would recommend adding `cluster_id` as that will allow for joins on the cluster table eventually.

Reply:
There is a separate design for the tables and `cluster_name` will not be in the tables. Currently collector does not know the name of the cluster that it is running on. The purpose of the `cluster_name` field is to let collector know the name of the cluster that it is running on. Collector can then use the part of the configuration that is relevant to it. There was an internal discussion within the collector team if central should just send the part of the configuration that is relevant to each sensor (Potentially each sensor would receive different parts of the configuration), or if the entire configuration would be sent to each sensor along with the cluster name and then collector would determine which part of the configuration is relevant to it. We went with the later for now, but it is possible that design will change. In either case `cluster_id` does not need to be a part of the protobuf definition here.

See the design for the table here https://docs.google.com/document/d/18z9tGG6ETIToKqcwBenTy4SWTGmeIj5xT7YXrSl9AF4/edit#heading=h.aaghmngw6yqj

There are also two draft PRs for the tables, though these need to be updated to be in agreement with the design document

https://github.com/stackrox/stackrox/pull/10723
https://github.com/stackrox/stackrox/pull/10743

Reply:
OK so this was not so much a `storage` message as it is an `api` message?

Reply:
Yes. The PR no longer adds anything to the `storage` directory.